Update model-card for Autofomer (#37231)
huggingface/transformers: docs/source/en/model_doc/autoformer.md

-->

<div style="float: right;">
    <div class="flex flex-wrap space-x-1">
        <img alt="PyTorch" src="https://img.shields.io/badge/PyTorch-DE3412?style=flat&logo=pytorch&logoColor=white">
    </div>
</div>

# Autoformer

> [Autoformer](https://huggingface.co/papers/2106.13008) is a time series forecasting model that rethinks Transformers with a twist — instead of just using attention, it brings in a **decomposition-based architecture** and a new **Auto-Correlation mechanism** to capture complex patterns in time series data.
> 
> What makes Autoformer unique? It doesn't treat trend and seasonal patterns as noise — it learns to **decompose** them, improving both performance and interpretability. Plus, it's super efficient for **long-term forecasting**, avoiding the bottlenecks of traditional Transformer-based approaches.
> 
> It’s especially great for scenarios like energy consumption, traffic flow, economic indicators, weather patterns, and disease spread forecasting.

Reviewer: ```suggestion
[Autoformer](https://huggingface.co/papers/2106.13008) is a model designed for long-term forecasting. It modifies the Transformer architecture with decomposition blocks that breaks down long-term trends, and it also adds an Auto-Correlation mechanism in place of self-attention to discover repeating patterns. The model combines these two designs to make better predictions.
```

Author: Done

> You can explore the model on the [Autoformer Hub Page](https://huggingface.co/elisim/autoformer-energy) or other community-hosted variants.

Reviewer: ```suggestion
You can find all the original Autoformer checkpoints on the [Hub](https://huggingface.co/models?search=autoformer).
```

Author: Done

> [!TIP]
> Click on the Autoformer model cards in the right sidebar to explore more examples and use cases like weather or energy time series forecasting.

> The example below demonstrates how to generate text with [`Pipeline`], [`AutoModel`]

Reviewer: ```suggestion
The example below demonstrates how to make a prediction with [`AutoformerForPrediction`].
```

Reviewer: Not resolved yet! I think its better to leave it as how to make a prediction since the docs don't explicitly show how to use it for things like weather and energy

<hfoptions id="usage">

> <hfoption id="Pipeline">

Reviewer: Lets remove this since the `Pipeline` doesn't support forecasting

Author: Done

```py
from transformers import pipeline

forecast = pipeline(task="time-series-forecasting", model="elisim/autoformer-energy")
output = forecast("energy.csv")  # this assumes a supported dataset format
```
</hfoption> 

> <hfoption id="AutoModel">

Reviewer: ```suggestion
<hfoption id="AutoformerForPrediction">
```

Author: Done

```py

> import torch

Reviewer: Lets use this example here:

```py
import torch
from huggingface_hub import hf_hub_download
from transformers import AutoformerForPrediction, AutoformerConfig

file = hf_hub_download(
    repo_id="hf-internal-testing/tourism-monthly-batch", filename="train-batch.pt", repo_type="dataset"
)
batch = torch.load(file)

model = AutoformerForPrediction.from_pretrained("huggingface/autoformer-tourism-monthly")

outputs = model.generate(
    past_values=batch["past_values"],
    past_time_features=batch["past_time_features"],
    past_observed_mask=batch["past_observed_mask"],
    static_categorical_features=batch["static_categorical_features"],
    future_time_features=batch["future_time_features"],
)

mean_prediction = outputs.sequences.mean(dim=1)
print(f"Mean prediction: {mean_prediction}")
```

Author: Updated the example. Done

from transformers import AutoformerForPrediction, AutoformerConfig

model = AutoformerForPrediction.from_pretrained("elisim/autoformer-energy")
config = AutoformerConfig.from_pretrained("elisim/autoformer-energy")

# Autoformer expects inputs like encoder_input: [batch_size, context_len, input_dim]
# Depending on how the model is implemented, check for the exact expected input name

dummy_input = torch.rand(1, config.seq_len, config.input_size)

# Wrap the tensor in a dictionary, assuming input is named "past_values" or similar
output = model(past_values=dummy_input)  # Replace with correct input name if different

print(output)
```
</hfoption> 

> <hfoption id="transformers-cli">

Reviewer: Also remove this since its unsupported

Author: Done

```bash
transformers-cli env  # transformers-cli doesn't currently support time series inputs
```
</hfoption> 
</hfoptions>

> Quantization

Reviewer: Lets also remove the Quantization content and Attention Mask Visualization stuff since none of that is supported either

Author: Done

Reviewer: This hasn't been removed yet!

Since Autoformer is lightweight and not extremely parameter-heavy like LLMs, quantization isn't typically needed. However, if you're experimenting with edge deployment or want to reduce size even more, standard PyTorch quantization techniques (like dynamic quantization for linear layers) can be applied.

```py
from torch.ao.quantization import quantize_dynamic
from transformers import AutoformerForPrediction

model = AutoformerForPrediction.from_pretrained("elisim/autoformer-energy")
quantized_model = quantize_dynamic(model, {torch.nn.Linear}, dtype=torch.qint8)
```

Attention Mask Visualization
Autoformer currently does not use standard attention mechanisms, so it isn't compatible with AttentionMaskVisualizer. Instead, it uses an Auto-Correlation mechanism, which doesn't operate on per-token attention masks.

## Notes

> - Autoformer was proposed in the paper [Autoformer: Decomposition Transformers with Auto-Correlation for Long-Term Series Forecasting](https://arxiv.org/abs/2106.13008) by Haixu Wu et al.
> - The model was contributed by [@elisim](https://huggingface.co/elisim) and [@kashif](https://huggingface.co/kashif).
> - The official implementation is available on [GitHub](https://github.com/thuml/Autoformer).

Reviewer: Lets also remove these `## Notes` since they aren't directly related to usage.

```suggestion
```

> - Check out Hugging Face’s blog post on this model: [Yes, Transformers are Effective for Time Series Forecasting (+ Autoformer)](https://huggingface.co/blog/autoformer).

Reviewer: ```suggestion
- Read the [Yes, Transformers are Effective for Time Series Forecasting (+ Autoformer)](https://huggingface.co/blog/autoformer) blog post for more details about using Autoformer to make predictions.
```

## AutoformerModel

[[autodoc]] AutoformerModel

> - forward

Reviewer: These don't need to be changed

Author: Reverted. Done
